mockbroker.go

```diff
@@ -207,19 +232,24 @@ func (b *MockBroker) handleRequests(conn net.Conn, idx int, wg *sync.WaitGroup)
 			b.serverError(err)
 			break
 		}
-		if len(encodedRes) == 0 {
-			continue
+		if len(encodedRes) != 0 {
+			binary.BigEndian.PutUint32(resHeader, uint32(len(encodedRes)+4))
+			binary.BigEndian.PutUint32(resHeader[4:], uint32(req.correlationID))
+			if _, err = conn.Write(resHeader); err != nil {
+				b.serverError(err)
+				break
+			}
+			if _, err = conn.Write(encodedRes); err != nil {
+				b.serverError(err)
+				break
+			}
+			b.lock.Lock()
+			requestResponse.ResponseSize = len(resHeader) + len(encodedRes)
```

Comment:
It would be if the responses are served in serial order but I do not know if this is always the case (third parties using the `MockBroker` concurrently).
But the cleaner implementation through a handler should discard that detail I believe.